libbeat/tests/system/test_cmd.py

```diff
@@ -132,5 +133,40 @@ def test_export_template(self):
         assert self.log_contains('"mockbeat-9.9.9-*"')
         assert self.log_contains('"codec": "best_compression"')
 
+    @unittest.skipUnless(INTEGRATION_TESTS, "integration test")
+    @attr('integration')
+    def test_test_output(self):
+        """
+        Test test output works
+        """
+        self.render_config_template("mockbeat",
+                                    os.path.join(self.working_dir,
+                                                 "mockbeat.yml"),
+                                    elasticsearch={"hosts": '["{}"]'.format(self.get_host())})
+        exit_code = self.run_beat(
+            extra_args=["test", "output"],
+            config="mockbeat.yml")
+
+        assert exit_code == 0
+        assert self.log_contains('parse url... OK')
+        assert self.log_contains('TLS... WARN secure connection disabled')
+        assert self.log_contains('talk to server... OK')
+
+    def test_test_wrong_output(self):
+        """
+        Test test wrong output works
+        """
+        self.render_config_template("mockbeat",
+                                    os.path.join(self.working_dir,
+                                                 "mockbeat.yml"),
+                                    elasticsearch={"hosts": '["badhost:9200"]'})
+        exit_code = self.run_beat(
+            extra_args=["test", "output"],
+            config="mockbeat.yml")
+
+        assert exit_code == 1
+        assert self.log_contains('parse url... OK')
+        assert self.log_contains('dns lookup... ERROR lookup badhost: no such host')
```

Comment:
I would be careful with matching the specific error message coming from the stdlib or the OS. This has bit me a few times when running my tests on different OSes and Golang versions.

```
elasticsearch: http://badhost:9200...
  parse url... OK
  connection...
    parse host... OK
    dns lookup... ERROR lookup badhost on 127.0.0.11:53: no such host
```

```
======================================================================
FAIL: Test test wrong output works
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/go/src/github.com/elastic/beats/libbeat/tests/system/test_cmd.py", line 169, in test_test_wrong_output
    assert self.log_contains('dns lookup... ERROR lookup badhost: no such host')
```